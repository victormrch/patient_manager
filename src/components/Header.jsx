function Header() {
  return (
    <h1 className="font-black text-5xl text-center mx-auto md:w-2/3">
      <span className="text-indigo-700">Advance</span> {""}
      Veterinary Care
    </h1>
  );
}

export default Header;
