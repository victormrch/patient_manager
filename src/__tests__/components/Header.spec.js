import React from "react";
import { render, screen } from "@testing-library/react";
import Header from "../../components/Header";
import "@testing-library/jest-dom/extend-expect";

describe("Header component specs", () => {
  it("should render correctly using snapshot testing", () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should display the header", () => {
    render(<Header />);

    const element = screen.getByRole("heading", {
      name: "Advance Veterinary Care",
    });

    expect(element).toBeInTheDocument();
    expect(element.tagName).toBe("H1");
    expect(element.textContent).toBe("Advance Veterinary Care");
  });
});
