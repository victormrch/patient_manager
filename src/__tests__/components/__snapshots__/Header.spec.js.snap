// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component specs should render correctly using snapshot testing 1`] = `
<DocumentFragment>
  <h1
    class="font-black text-5xl text-center mx-auto md:w-2/3"
  >
    <span
      class="text-indigo-700"
    >
      Advance
    </span>
     Veterinary Care
  </h1>
</DocumentFragment>
`;
